funding/batch.go

```diff
@@ -416,7 +433,7 @@ func (b *Batcher) BatchFund(ctx context.Context,
 		// response on the update chan.
 		channel := channel
 		eg.Go(func() error {
-			return b.waitForUpdate(channel, false)
+			return b.waitForUpdate(ctx, channel, false)
```

Comment:
I am still unsure why @guggero's suggestion to use a similar `errgroup` context as above causes the unit and itests to flake. I've used the main context here for now which shouldn't cause any problems since a deadline context is derived from it in the `waitForUpdate` method.

Reply:
Interesting, were the flakes related to the opening flow or did they appear in other parts ?

Reply:
I've re-enabled the errgroup context to trigger the flake, it is related to the channel opening flow, see https://github.com/lightningnetwork/lnd/actions/runs/7693557479/job/20962556158?pr=8367#step:5:651. This passes locally.

Don't know yet why it is running into `Number of pending channels exceed maximum` if we use a context that's derived from the one it works with.

Reply:
But I think the test nees to be updated because it can easily happen that 3 of the 4 channel openings are registered for the peers only carol has this limit (`	carol := ht.NewNode("carol", []string{"--minchansize=200000"})
`)

Reply:
> This fails for me locally as well tho:
> 
> make itest icase=batch_channel_funding

True, succeeds with the global context, but fails with the error group context 🤔

Reply:
Not sure if I understand you correctly, but `eg.Wait()` waits for all spawned up goroutines until it proceeds (even if one goroutine encounters an error prematurely), so the only way to trigger an exit of all goroutines is an errorGroup to create the errGroup with the context and check for this context in all goroutines for the cancel deadline.

Reply:
you could implement a cleanup after the `batchopenwitherr` which abandons any pending channel resutling from the first try by alice for all peers ?

Reply:
Thanks for the analysis @ziggie1984. I'd also prefer the use of the `errgroup.WithContext()`. Although the problem of exceeding # of pending channels seems to happen during manual testing in `regtest` as well where I delay some calls so that I can `Ctrl+C` them. If I do that then I reliably run into the error `exceeding pending channels`. Only after a restart seems the remote side to clean up the pending channels, and I can open again. This doesn't sound like acceptable behavior?

Reply:
The patch I am testing the delays with:
```
diff --git a/funding/batch.go b/funding/batch.go
index 745f2fd67..dae31be42 100644
--- a/funding/batch.go
+++ b/funding/batch.go
@@ -26,7 +26,7 @@ const (
        // funding operation. If the whole operation takes longer than this
        // timeout, it will be aborted. During the timeout period the user can
        // send Ctrl+C to the process to abort the funding process manually.
-       DefaultBatchFundingTimeout = 60 * time.Second
+       DefaultBatchFundingTimeout = 10 * time.Second
 )
 
 var (
diff --git a/funding/manager.go b/funding/manager.go
index 42aa7bc12..57bd27d0c 100644
--- a/funding/manager.go
+++ b/funding/manager.go
@@ -4354,6 +4354,7 @@ func (f *Manager) announceChannel(localIDKey, remoteIDKey *btcec.PublicKey,
 // to initiate a single funder workflow with the source peer.
 // TODO(roasbeef): re-visit blocking nature..
 func (f *Manager) InitFundingWorkflow(msg *InitFundingMsg) {
+       time.Sleep(time.Second * 5)
        f.fundingRequests <- msg
 }
```

Reply:
Yes current behavior is not good:

Let's do this maybe ?

In the cleanup message add another section which loops through the reservations and sends an error to the peer so an add on here: https://github.com/lightningnetwork/lnd/blob/master/funding/batch.go#L535-L544

which for all reservations which where part of this opening but still had NO funding intent we also call:
```
		f.failFundingFlow(resCtx.peer, cid, cause)

```

Reply:
basically for all openings which exit early before even receiving their ACCEPT_CHANNEL_MSG, we need to do this because this will trigger the deletion of the activereservations of our peer (at least for lnd)